### PR TITLE
Glimmerize KeyValueHeader

### PR DESCRIPTION
**Pull request by Monkeychip**
Using this component in the PKI redesign work, and figured it was good practice to glimmerize while I could.

This is the breadcrumb component (poorly named 🤔). I could be talked into renaming it, but it's used in a LOT of places, so not sure it's worth it.

image

Notes:
- Tested on Enterprise locally, and works great.

### CODE REVIEW

File: ui/lib/core/addon/components/key-value-header.js [hunk 1]
```diff
@@ -1,87 +1,69 @@
...
 import utils from 'vault/lib/key-utils';
-import layout from '../templates/components/key-value-header';
 import { encodePath } from 'vault/utils/path-encoding-helpers';
 
-export default Component.extend({
-  layout,
-  tagName: 'nav',
-  classNames: 'key-value-header breadcrumb',
-  ariaLabel: 'breadcrumbs',
-  attributeBindings: ['ariaLabel:aria-label', 'aria-hidden'],
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/key-value-header.js:12]: From what I understand about attribute bindings this read as a conditional. Basically by default aria-label was always attached as an attribute unless `airaLabel` was false. See glimmer [docs here](https://guides.emberjs.com/release/upgrading/current-edition/glimmer-components/). I couldn't find any cases where aria-label should instead by aria-hidden.

Comment (Monkeychip) [reply ui/lib/core/addon/components/key-value-header.js:12]: Note: we followed the classNameBindings pattern in the old version, so it maybe helpful to look at that example on the docs

Comment (Monkeychip) [reply ui/lib/core/addon/components/key-value-header.js:12]: This is now what I have in the .hbs template:
`<nav class="key-value-header breadcrumb" aria-label="breadcrumbs">`

File: ui/lib/core/addon/components/key-value-header.hbs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<nav class="key-value-header breadcrumb" aria-label="breadcrumbs">
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/key-value-header.hbs:1]: same component, but added the wrapper `<nav with the class and aria-label...>`

File: ui/lib/core/addon/components/key-value-header.js (hunk 1 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/key-value-header.js:12]: This looks right to me. I didn't see anywhere in the code that we passed in `aria-hidden` as an attribute for this component

File: ui/lib/core/addon/components/key-value-header.js [hunk 3]
```diff
@@ -1,87 +1,69 @@
...
+/**
+ * @module KeyValueHeader
+ * KeyValueHeader components show breadcrumbs.
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/key-value-header.js:7]: Thanks for adding the `jsdoc` - does it just show breadcrumbs? Or is it breadcrumbs specifically for the secrets engines?

File: ui/lib/core/addon/components/key-value-header.js [hunk 4]
```diff
@@ -1,87 +1,69 @@
...
+ *
+ * @example
+ * ```js
+ * <KeyValueHeader @requiredParam={requiredParam} @optionalParam={optionalParam} @param1={{param1}}/>
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/key-value-header.js:11]: Pulling a real example from somewhere in the code might be a little more explanatory. It'd be helpful for me at least to see an example of what this looks like in practice!

File: ui/lib/core/addon/components/key-value-header.js [hunk 5]
```diff
@@ -1,87 +1,69 @@
...
+ * ```
+ * @param {string} [mode=null] - Used to set the currentPath.
+ * @param {string} [baseKey=null] - Used to generate the path backward.
+ * @param {string} [currentPath=null] - requiredParam is...
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/key-value-header.js:15]: I think this one may have been missed

File: ui/lib/core/addon/components/key-value-header.js [hunk 6]
```diff
@@ -1,87 +1,69 @@
...
-          path: path,
-          model: ancestor,
-        });
-      });
+    if (!baseKey) {
+      return crumbs;
+    }
 
+    const path = this.args.path;
+    const currentPath = this.currentPath;
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/key-value-header.js:59]: small nit: are the variable declarations on lines 58-60 helpful here? 🤔  I might be alone thinking this - but readability might be easier knowing throughout this method where each variable is coming from further down (i.e. `baseKey` is a `const` defined within this method, but `currentPath` is actually a getter for the component) - it doesn't seem like we save a ton of cleanup by renaming them to variables. just a thought!

**Review by hellobontempo — APPROVED** [review]
Nice job! Just a few cleanup comments, nothing blocking 🎸

File: ui/lib/core/addon/components/key-value-header.js (hunk 3 above)
Comment (Monkeychip) [reply ui/lib/core/addon/components/key-value-header.js:7]: It really only shows breadcrumbs. The hbs template is essentially two links separated by the > icon. Regarding if it's specific for secret-engines. The component is used only in secret engines, however, the styling of the component (e.g. the class) is used outside of secrets engine. example, for the class "key-value-header" it's used in replication.

File: ui/lib/core/addon/components/key-value-header.js (hunk 4 above)
Comment (Monkeychip) [reply ui/lib/core/addon/components/key-value-header.js:11]: Ah, yep, I forgot this bit. Copy and pasted.

File: ui/lib/core/addon/components/key-value-header.js (hunk 5 above)
Comment (Monkeychip) [reply ui/lib/core/addon/components/key-value-header.js:15]: Sure was. Thanks.

File: ui/lib/core/addon/components/key-value-header.js (hunk 6 above)
Comment (Monkeychip) [reply ui/lib/core/addon/components/key-value-header.js:59]: Agreed. I struggle with this on glimmerizing. How much do I refactor as well as glimmer? This was a direct copy from the previous function (I didn't change any of the functionality). I realize the change here is very low risk, but I've been bit a couple times combing refactoring and glimmer together. Again, really not sure. Let me sit with this until tomorrow and I'll take another look. It's not a big deal, it's more how do I want to think about this as a rule going forward.

File: ui/lib/core/addon/components/key-value-header.js (hunk 3 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/key-value-header.js]: gotcha - thanks!

**Pull request merged**